agent/config/flags.go

```diff
@@ -75,6 +75,7 @@ func AddFlags(fs *flag.FlagSet, f *BuilderOpts) {
 	add(&f.Config.HTTPConfig.AllowWriteHTTPFrom, "allow-write-http-from", "Only allow write endpoint calls from given network. CIDR format, can be specified multiple times.")
 	add(&f.Config.EncryptKey, "encrypt", "Provides the gossip encryption key.")
 	add(&f.Config.Ports.GRPC, "grpc-port", "Sets the gRPC API port to listen on (currently needed for Envoy xDS only).")
+	add(&f.Config.HTTPConfig.MaxHeaderBytes, "http-max-header-bytes", "Sets the HTTP Server's Max Header Bytes.")
```

Comment:
It is possible to specify any config option as a CLI flag using `-hcl='...'`. 

In general we'd like to reduce the number of CLI flags so I don't think we should add this one here. Keeping the new setting as only a config field and allowing someone to set it with `-hcl='MaxHeaderBytes = value'`.

Reply:
Understood.  This has been removed.